Xbim.Common/Step21/Part21Formatter.cs

```diff
@@ -32,28 +32,29 @@ public string Format(string fmt, object arg, IFormatProvider formatProvider)
         {
             // Convert argument to a string           
 
-            if (!string.IsNullOrEmpty(fmt) && fmt.ToUpper() == "R" && arg is double)
+            if (!string.IsNullOrEmpty(fmt) && fmt.ToUpper() == "G" && arg is double)
             {
                 var dArg = (double)arg;
+                return dArg.ToString("G", StepText.DoubleCulture);
 
                 // if compiler flag, only then do the following 3 lines
           
-                var rDoubleStr = dArg.ToString("R", StepText.DoubleCulture);
-                var fixedDbl = double.Parse(rDoubleStr, StepText.DoubleCulture);
-                var result = fixedDbl.ToString("R", StepText.DoubleCulture);
+                //var rDoubleStr = dArg.ToString("R", StepText.DoubleCulture);
+                //var fixedDbl = double.Parse(rDoubleStr, StepText.DoubleCulture);
+                //var result = fixedDbl.ToString("R", StepText.DoubleCulture);
 
                 //decimal decArg = new Decimal(dArg);                                
                 // string result = decArg.ToString().ToUpper();
                 // string result = string.Format("{0:e22}", arg);
                 //string result = dArg.ToString("R", CultureInfo.CreateSpecificCulture("en-US"));
-                if (result.Contains(".")) return result;
+                //if (result.Contains(".")) return result;
```

Comment:
It is not possible to comment out this and following lines. Many numbers would than be wrong.

Reply:
Yes, now I see that we need those lines to get integer number with dot in the end. Just uncommented this.